CHANGELOG.rst

```diff
@@ -1,6 +1,77 @@
+- Plugin systems allowing third party libraries to extend MLflow functionality. The `proposal document <https://gist.github.com/zblz/9e337a55a7ba73314890be68370fa69a>`_ gives the full detail of the three main changes: 
+
+  - Additional providers of tracking stores can be registered using the ``mlflow.tracking_store`` entrypoint. (#881, @zblz)
+  - Additional providers of artifact repository can be registered using the ``mlflow.artifact_repository`` entrypoint. (#882, @mociarain)
+  - The logic generating run metadata from the run context (e.g. ``source_name``, ``source_version``) has been refactored into an extendable system of run context providers. Plugins can register additional providers using the `mlflow.run_context_provider` entrypoint, which add to or overwrite tags set by the base library. (#913, #926, #930, #978, @acroz)
+
+- Support for authentication to the tracking server in the R client. Now R users can connect to secure tracking servers via credentials set in environment variables, or provide custom plugins for setting the credentials. As an example, in this release is the Databricks plugin which can detect existing Databricks credentials to allow users to connect to the Databricks tracking server. (#938, #959, #992, @tomasatdatabricks)
+
+
+Breaking changes:
+
+- [Scoring] Use pandas-split format as default for JSON requests to pyfunc scoring server (#960, @dbczumar)
+- [Scoring] When reading pandas dataframes from json, do not infer data types (#916, @mparkhe)
+
+
+More features and improvements:
+
+- [UI] Add a button for downloading artifacts (#967, @mateiz)
+- [CLI] Add ``mlflow runs`` CLI utility (#720, @DorIndivo)
+- [CLI] ``MLFLOW_EXPERIMENT_NAME`` env variable and ``--experiment-name`` arguments for CLI command (#889, #894, @mparke)
+- [Examples] Add Image classification example with Keras. (#743, @tomasatdatabricks )
+- [R][Runs] Allow client to infer context info when creating new run in fluent API (#958, @tomasatdatabricks)
+- [Logging] LogBatch support (#950, #951, #955, @smurching)
+- [Artifacts] Add ``get_artifact_uri()`` and ``_download_artifact_from_uri`` convenience functions (#779)
+- [Artifacts] Allow writing Spark models directly to the target artifact store when possible (#808, @smurching)
+- [Models] Make mlflow.pytorch.pickle_module the default when saving PyTorch models (#861)
+- [Models] Add ``code paths`` parameter to pytorch model persistence functions (#842, dbczumar)
+- [Models] Add CloudPickle serialization module for PyTorch model persistence (#851, @dbczumar)
+- [Projects] Add support for Gitlab and Bitbucket project repo url (#901)
+- [Search] Support for Search Runs API using "filter" string (#905, @mparke)
+- [Search] Search API: allowing param value to have any content for checks (#788, @mparkhe)
```

Comment:
These might be major enough to be a top level feature. Are they public APIs now, or is this supported in the text field in the UI?

Reply:
@mparkhe is the search stuff in open source done enough to aggregate and summarize as a major feature? Or is there more work to be done?

Reply:
Not at this point. IMO, let's not highlight this as a high level feature quite yet. We have some pieces in place but not them all. We still need

- UI support for searching with tags
- UI support flexibility to search for metric/param/tag keys with spaces and some special characters.
- Front-end code hydrating `filter` field in proto